packages/nx/src/plugins/js/project-graph/build-dependencies/explicit-package-json-dependencies.ts

```diff
@@ -99,10 +119,26 @@ function processPackageJson(
 }
 
 function readDeps(packageJson: PackageJson) {
-  return [
-    ...Object.keys(packageJson?.dependencies ?? {}),
-    ...Object.keys(packageJson?.devDependencies ?? {}),
-    ...Object.keys(packageJson?.peerDependencies ?? {}),
-    ...Object.keys(packageJson?.optionalDependencies ?? {}),
+  const deps = {};
+
+  /**
+   * We process dependencies in a rough order or increasing importance such that if a dependency is listed in multiple
```

Comment:
```suggestion
   * We process dependencies in a rough order of increasing importance such that if a dependency is listed in multiple
```

Will apply once the first CI run completes to see if there are other things to address